cgnet/simplify.py

```diff
@@ -0,0 +1,33 @@
+import networkx as nx
+
+def is_simple_node( graph, node ):
+    """A node is "Simple" if none of the following is true
+    - it has multiple inputs (it joins chains together)
+    - it has no inputs (it's a root node)
+    - it has multiple outputs (it splits chains apart)
+    - it has no outputs (it's a leaf node)
+
+    Keyword arguments:
+    node -- A networkx DiGraph Node
+    """
+    return graph.in_degree(node) == 1 \
+            and graph.out_degree(node)  == 1
+
+def simplify_graph( graph ):
```

Comment:
I think I'd name this something like `simplified` to emphasize that we aren't modifying the original graph.